nms/app/packages/magmalte/app/components/context/RefreshContext.js

```diff
@@ -0,0 +1,305 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @flow strict-local
+ * @format
+ */
+'use strict';
+
+import MagmaV1API from '@fbcnms/magma-api/client/WebClient';
+import {useContext, useEffect, useState} from 'react';
+
+type Props = {
+  context: typeof React.Context,
+  networkId: string,
+  type: refreshType,
+  interval?: number,
+  id?: string,
+  //$FlowIgnore
+  refreshingState?: any,
+  enqueueSnackbar?: (msg: string, cfg: {}) => ?(string | number),
+};
+
+type refreshType = 'subscriber' | 'gateway' | 'enodeb';
+
+export function useRefreshingContext(props: Props) {
+  const ctx = useContext(props.context);
+  const [refreshingState, setRefreshingState] = useState(ctx);
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      fetchRefreshState({
+        type: props.type,
+        networkId: props.networkId,
+        refreshingState: refreshingState,
+        id: props.id,
+        enqueueSnackbar: props.enqueueSnackbar,
+      }).then(response => {
+        if (response) {
+          setRefreshingState(response);
+        }
+      });
+    }, props.interval);
+
+    return () => {
+      // Update context on leave
+      clearInterval(intervalId);
+      updateContext({
+        context: ctx,
+        type: props.type,
+        refreshingState: refreshingState,
+        networkId: props.networkId,
+      });
+    };
+  }, [
+    ctx,
+    props.enqueueSnackbar,
+    props.id,
+    props.interval,
+    props.networkId,
+    props.type,
+    refreshingState,
+  ]);
+  return refreshingState;
+}
+
+export function updateContext(props: Props) {
+  if (props.refreshingState) {
+    if (props.type === 'subscriber') {
+      props.context.setState(
+        null,
+        props.refreshingState.state,
+        props.refreshingState.sessionState,
+      );
+    } else {
+      props.context.setState(
+        null,
+        props.networkId,
+        props.refreshingState.state,
+      );
+    }
+  }
+}
+
+type fetchProps = {
```

Comment:
Can you move the fetch part of the code under app/state directory. Also this logic seems duplicated, we should merge the Init and the fetch methods.